Vacation-STUDENT_ID_300506334/apps/api/src/admin/vacations/businessLogic.ts

```diff
@@ -0,0 +1,115 @@
+import { ISignToken } from "../../auth/signTokenFn";
+import { getConnection } from "../../db";
+import { toLocal } from "../../helpers/currentDateTimeFn";
+import { IcreateVacation } from "../../middlewares/joiValidations";
+import { IvacationDB } from "../../vacations/businessLogic";
+
+type Id = number
+
+interface IUpdateVacation {
+  id: number;
+  description?: string;
+  destination?: string;
+  image?: string;
+  start_date_time?: object;
+  end_date_time?: object;
+  price_usd?: number;
+};
+
+interface IFollowersPerVacationResponseObj {
+  destination: string;
+  number_of_followers: number;
+};
+
+async function updateVacation(
+  reqBody: IUpdateVacation,
+  reqUserData: ISignToken
+  ): Promise<void>{
+  //vacation id mandatory 
+  // in this function after I have at least one param to update
+  const { id, description, destination, image, start_date_time, end_date_time , price_usd } = reqBody;
+const updatedDateTime: string = toLocal( new Date()).slice(0, 19).replace('T', ' ')
+const arrBodyKeys: Array<string> = Object.keys(reqBody)
+const newBody: IUpdateVacation | {} = arrBodyKeys.reduce((accum, key) => {
+ if (Boolean(reqBody[key])) accum[key] = reqBody[key];
+  return accum;
+}, {});
+  const paramsArr: Array<any> = Object.values(newBody);
```

Comment:
better to split this function to some query builder.
getQuery()

Reply:
no problem, will do